python/paddle/fluid/layers/tensor.py

```diff
@@ -97,10 +99,28 @@ def create_parameter(shape,
             import paddle.fluid.layers as layers
             W = layers.create_parameter(shape=[784, 200], dtype='float32')
     """
+    check_type(shape, 'shape', (list, tuple, numpy.ndarray), 'create_parameter')
+    for item in shape:
+        if six.PY2:
+            check_type(item, 'item of shape',
+                       (int, long, numpy.uint8, numpy.int8, numpy.int16,
+                        numpy.int32, numpy.int64), 'create_parameter')
+        else:
+            check_type(item, 'item of shape',
+                       (int, numpy.uint8, numpy.int8, numpy.int16, numpy.int32,
+                        numpy.int64), 'create_parameter')
+
+    check_dtype(dtype, 'dtype', ['float16', 'float32', 'float64'],
```

Comment:
Is it a good choice to restrict parameters to be 'float16', 'float32', 'float64'

Reply:
Done. Thanks. If the `dtype` is `int`, then it's necessary to set default_initializer and this works fine.